pyquil/api/qvm.py

```diff
@@ -313,8 +313,15 @@ def expectation(self, prep_prog, operator_programs=None, needs_compilation=False
             sample. The expectations returned from *different* ``expectation`` calls *will then
             generally be different*.
 
+        To measure the expectation of a PauliSum, you probably want to
+        do something like this::
+
+                progs, coefs = hamiltonian.get_programs()
+                expect_coeffs = np.array(cxn.expectation(prep_program, operator_programs=progs))
+                return np.real_if_close(np.dot(coefs, expect_coeffs))
+
         :param Program prep_prog: Quil program for state preparation.
-        :param list operator_programs: A list of PauliTerms. Default is Identity operator.
+        :param list operator_programs: A list of Programs. Default is Identity operator.
```

Comment:
I know this may be beyond the scope of the PR, but passing programs to represent observables is just seems like a hack, and is bound to lead to confusion -- it certainly caught me by surprise.

Given this highly counterintuitive representation of observables, I suggest that the doc also include an example, e.g., PauliTerm('X'.0).program, as the input

But the real fix should to be stop representing observales with programs.  I don't think that is a defensible design choice.

Reply:
Agree. We'll leave #261 open, but I think this PR is a strictly improving change